lib/edit-site-page.php

```diff
@@ -74,11 +79,53 @@ function gutenberg_edit_site_init( $hook ) {
 		$settings['fontSizes'] = $font_sizes;
 	}
 
-	// Get root template by trigerring `./template-loader.php`'s logic.
+	// Get all templates by triggering `./template-loader.php`'s logic.
+	$template_getters  = array(
+		'get_embed_template',
+		'get_404_template',
+		'get_search_template',
+		'get_home_template',
+		'get_privacy_policy_template',
+		'get_post_type_archive_template',
+		'get_taxonomy_template',
+		'get_attachment_template',
+		'get_single_template',
+		'get_page_template',
+		'get_singular_template',
+		'get_category_template',
+		'get_tag_template',
+		'get_author_template',
+		'get_date_template',
+		'get_archive_template',
+	);
+	$template_ids      = array();
+	$template_part_ids = array();
+	foreach ( $template_getters as $template_getter ) {
+		call_user_func( $template_getter );
+		apply_filters( 'template_include', null );
+		if ( isset( $_wp_current_template_id ) ) {
+			$template_ids[ $_wp_current_template_name ] = $_wp_current_template_id;
+		}
+		if ( isset( $_wp_current_template_part_ids ) ) {
+			$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
+		}
+		$_wp_current_template_id        = null;
+		$_wp_current_template_name      = null;
+		$_wp_current_template_content   = null;
+		$_wp_current_template_hierarchy = null;
+		$_wp_current_template_part_ids  = null;
+	}
```

Comment:
It would be cool to extract all the logic here into a utility function `gutenberg_get_available_templates` or something like that. this bootstrap is becoming too complex.

Reply:
Good idea. It might be useful for plugin developers as well.

Reply:
Should we do this as a follow-up?

Reply:
Yes